src/__tests__/nodeSuites/ip-addresses-setting.spec.js

```diff
@@ -52,44 +54,52 @@ const configWithIPAddressesDefault = {
 
 export default function ipAddressesSettingAssertions(mock, assert) {
 
-  let splitioEnabled = SplitFactory(configWithIPAddressesEnabled);
-  let clientEnabled = splitioEnabled.client();
-  const settingsEnabled = SettingsFactory(configWithIPAddressesEnabled);
-
-  mock.onPost(settingsEnabled.url('/events/bulk')).replyOnce(req => {
-    assert.equal(HEADER_SPLITSDKMACHINEIP in req.headers, true, `Request must include ${HEADER_SPLITSDKMACHINEIP} header.`);
-    assert.equal(HEADER_SPLITSDKMACHINENAME in req.headers, true, `Request must include ${HEADER_SPLITSDKMACHINENAME} header.`);
-    clientEnabled.destroy();
-    return [200];
-  });
-
-  clientEnabled.track('nicolas@split.io', 'sometraffictype', 'someEvent', 10);
+  const finish = (function*() {
+    const clients = [];
+    let count = 1;
+    do{
+      const currentYield = yield;
+      if(currentYield) {
+        clients.push(currentYield);
+        count--;
+      }else
+        count++;
+    }while(count);
+    clients.forEach(client => client.destroy());
+    assert.end();
+  })();
 
-  let splitioDisabled = SplitFactory(configWithIPAddressesDisabled);
-  let clientDisabled = splitioDisabled.client();
-  const settingsDisabled = SettingsFactory(configWithIPAddressesDisabled);
+  function assertHeaders(IPAddressesEnabled, req){
+    assert.equal(HEADER_SPLITSDKMACHINEIP in req.headers, IPAddressesEnabled, `Request must include ${HEADER_SPLITSDKMACHINEIP} header if IPAddressesEnabled is true.`);
```

Comment:
Truly a minor thing but the assertion message is missleading (it always talks about the value being true). Maybe we can have something like:
`Request must ${IPAddressesEnabled ? '' : 'NOT '}include ${HEADER_SPLITSDKMACHINEIP} header if IPAddressesEnabled is ${IPAddressesEnabled}.`

Reply:
Done